src/Tools/dotnet-trace/CommandLine/Commands/CollectCommand.cs

```diff
@@ -39,11 +39,16 @@ private static async Task<int> Collect(CancellationToken ct, IConsole console, i
                 Debug.Assert(output != null);
                 Debug.Assert(profile != null);
                 Console.Clear();
-                if (processId <= 0)
+                if (processId < 0)
                 {
                     Console.Error.WriteLine("Process ID should not be negative.");
                     return ErrorCodes.ArgumentError;
                 }
+                else if (processId == 0)
+                {
+                    Console.Error.WriteLine("--process-id is required");
```

Comment:
nit: I agree with @josalem here, where validation should happen in the `System.CommandLine`